File: Algorithms/Sort/insertionSort.cpp
#include <bits/stdc++.h>
using namespace std;

void printArray(int arr[], int n)
{
    int i;
    for (i = 0; i < n; i++)
        cout << arr[i] << " ";
    cout << endl;
}
void insertionSort(int arr[],int length){
    int i , j , key ;
    for(i = 1 ;i<length ; i++){
        key = arr[i];
        j= i-1;
        while(j>=0 && arr[j] > key){
            arr[j+1] = arr[j];
            j = j-1;
        }
        arr[j+1] = key;
    }
}

int main(){
    int arr[]= { 2,12,5,7,9,10};
    int length = sizeof(arr) / sizeof(arr[0]);
    insertionSort(arr, length);
    printArray(arr, length);
    return 0;
}

File: Algorithms/Sort/insertionSort.js
const insertionSort = (arr) => {
    for(let i = 0 ; i< arr.length ; i++) {
        while(i > 0 && arr[i-1] > arr[i]) {
            const temp = arr[i-1]
            arr[i-1] = arr[i];
            arr[i] = temp
            i--
        }
    }
}

const arr = [2,4,5,2,3,1,0,23,43,12,8,43,20]
insertionSort(arr)
console.log(arr)